detectable/src/main/java/com/synopsys/integration/detectable/detectables/pip/parser/RequirementsFileExtractor.java

```diff
@@ -24,11 +29,44 @@ public RequirementsFileExtractor(
         this.requirementsFileDependencyTransformer = requirementsFileDependencyTransformer;
     }
 
-    public Extraction extract(File requirementsFileObject) throws IOException {
-//        String requirementsFileText = FileUtils.readFileToString(requirementsFileObject, StandardCharsets.UTF_8);
-        List<RequirementsFileDependency> dependencies = requirementsFileTransformer.transform(requirementsFileObject);
-        DependencyGraph dependencyGraph = requirementsFileDependencyTransformer.transform(dependencies);
-        CodeLocation codeLocation = new CodeLocation(dependencyGraph);
-        return Extraction.success(codeLocation);
+    public Set<File> findChildFileReferencesInParent(File parentRequirementsFile) throws IOException {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(parentRequirementsFile))) {
+            List<String> tokens;
+            Set<File> childRequirementsFiles = new HashSet<>();
+            String childFileReferenceToken;
+            Path childFileReferencePath;
+            File childFile;
+
+            for (String line; (line = bufferedReader.readLine()) != null; ) {
+                if (!line.startsWith("-r")) {
+                    continue;
+                }
+                tokens = Arrays.asList(line.split(" "));
+                if (tokens.size() > 1 && Objects.equals(tokens.get(0), "-r")) {
+                    childFileReferenceToken = tokens.get(1);
+                    if (!childFileReferenceToken.isEmpty()) {
+                        childFileReferencePath = Paths.get(childFileReferenceToken);
+                        childFile = childFileReferencePath.toFile();
+                        if (childFile.exists()) {
```

Comment:
We should consider logging a warning or similar if a referenced file does not exist.
Also, I wasn't able to find such examples based on my quick searches, but is it possible for the `-r` option to point to a file in a different directory? If it is possible, will this code still work as expected?

Reply:
Good question. I had the same one and PIP docs don't actually specify what type of path is to be provided in the parent requirements.txt -- absolute or relative. 

I think the parser should follow the same strategy the PIP CLI follows. 
I had verified that the PIP CLI works with absolute paths as references so above code works for absolute paths. So to answer your question, yes the above code will work for a file in another directory (even an external one) because the path is absolute.

Your comment though made me think of the scenario when relative paths are provided.
I did some tests and confirmed that PIP CLI does support relative paths too along with absolute paths (basically -r supports UNIX style paths). The parser should ideally support the same. I'll take a look at this part and confirm